[Pal/Linux-SGX] Fix renaming issue with protected files

svenkata9 opened the pull request:
## Description of the changes <!-- (reasons and measures) -->
With certain workloads involving Federated Learning, the frameworks create protected files and rename them as part of the flow. This does not work currently with protected files because every protected file maintains the path name in its metadata (stored when the file was first created) and which is checked during open to make sure the incoming path is the same as what is already stored in the metadata for this file.

This PR is to address this issue by handling rename for protected files. A normal rename call does not need the file to be open in RW mode, but since the metadata needs to be updated with the new path, we pass a special `PAL_OPTION_RENAME` flag to `DkStreamOpen` during rename and open the protected file in RW mode with required permissions, update the metadata with the new path and flush it. Then the flow goes through the one for normal file renaming. If the ocall to rename fails, we restore the metadata back to what it was earlier so the file remains usable still.

There is a new regression test in LibOS `pf_rename` to test this

<!--
    If your PR fixes an issue, please remember to add "Fixes #issue_number"
    here, to automatically close it on merge. -->

## How to test this PR? <!-- (if applicable) -->
The newly added regression test in LibOS `pf_rename` should pass.

Signed-off-by: Sankaranarayanan Venkatasubramanian <sankaranarayanan.venkatasubramanian@intel.com>

<!--
    Please fill in the following form before submitting this PR
    and ensure that your code follows our coding style guideline:
    https://graphene.readthedocs.io/en/latest/devel/coding-style.html -->

## Description of the changes <!-- (reasons and measures) -->

<!--
    If your PR fixes an issue, please remember to add "Fixes #issue_number"
    here, to automatically close it on merge. -->

## How to test this PR? <!-- (if applicable) -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/graphene/2681)
<!-- Reviewable:end -->

svenkata9 committed:
[Pal/Linux-SGX] Fix renaming issue with protected files

With certain workloads involving Federated Learning, the frameworks create protected files and
renames them as part of the flow. This does not work currently with protected files because every
protected file maintains the path name in its metadata (stored when the file was first created)
and which is checked during open to make sure the incoming path is the same as what is already
stored in the metadata for this file.

A normal rename call does not need the file to be open in RW mode, but since the metadata needs
to be updated with the new path, we pass a special PAL_OPTION_RENAME flag to DkStreamOpen during
rename and open the protected file in RW mode with required permissions, update the metadata with
the new path and flush it. Then the flow goes through the one for normal file renaming. If the
ocall to rename fails, we restore the metadata back to what it was earlier so the file remains
usable still.

Signed-off-by: Sankaranarayanan Venkatasubramanian <sankaranarayanan.venkatasubramanian@intel.com>

svenkata9 commented:
@duanbing I abandoned because of conflicts after rebasing (it was ~2 months old). Please use this one.

duanbing commented on Pal/src/host/Linux-SGX/db_files.c:
```diff
@@ -792,9 +803,59 @@ static int file_rename(PAL_HANDLE handle, const char* type, const char* uri) {
+            return -PAL_ERROR_DENIED;
+        }
         return unix_to_pal_error(ret);
     }
```
Need `free(tmp);` after renaming?

svenkata9 replied:
I didn't get your comment exactly. But, tmp needs to be freed up anyway before we return from this function. And, this rename is restoring the metadata back to what it was before. And, tmp is no more of use when we enter this if (ret < 0) condition.

duanbing replied:
@svenkata9 I mean heap variable `tmp` is not freed when renaming returns >= 0.

svenkata9 replied:
@duanbing It is assigned in this line, so cannot be freed.
 https://github.com/gramineproject/graphene/blob/33a68bc4302891e8c591570942bc39394acefd23/Pal/src/host/Linux-SGX/db_files.c#L806

duanbing replied:
Got it. Thanks.

dimakuv reviewed (changes requested):
> Reviewed 11 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/graphene/2681)* status: all files reviewed, 22 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @svenkata9)

*[a discussion](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yhSE7OocZItNkL4P:-Mj3yhSE7OocZItNkL4Q:b9wgu7z) (no related file):*
@pwmarcz will need to review this. Especially the part with the new `rename` test -- Pawel also added a similar test in his PR, so I think the two tests can be merged into one.

---
___
*[-- commits, line 5 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3pH2W93B098FdFtlV):*
`renames` -> `rename`
___
*[-- commits, line 15 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3p7yY-rO20zDfIwaQ):*
This doesn't seem to wrap on 72-char limit (the default limit for git messages). Could you re-wrap lines to 72 chars?
___
*[LibOS/shim/test/regression/.gitignore, line 77 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3pX96A2ETRkiVe4r_:-Mj3pX96A2ETRkiVe4ra:b-nnqa32) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/.gitignore#L77)):*
> ```ignore
> /openmp
> /pf_rename
> /pftmp
> ```

Please move `/pftmp` (and also `/tmp`) in a separate section in this file (i.e., separates from other files by empty lines). These two entries are directories for temporary files, not executables, so they don't belong in this list logically.
___
*[LibOS/shim/test/regression/pf_rename.c, line 14 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3pvgJB7TcKA0U1bRX:-Mj3pvgJB7TcKA0U1bRY:b-pah0c4) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L14)):*
> ```C
>     char buf[15];
>     char input_text[] = "Hello, world!";
>     int fd = open("pftmp/foo.txt", O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
> ```

Can you use macros for these two paths? Like this:
```
#define PF_FOO_PATH "pftmp/foo.txt"
#define PF_BAR_PATH "pftmp/bar.txt"
```
___
*[LibOS/shim/test/regression/pf_rename.c, line 16 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3q9MG5FpakVxcfCeH:-Mj3q9MG5FpakVxcfCeI:b-q056it) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L16)):*
> ```C
>     int fd = open("pftmp/foo.txt", O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
>     if (fd < 0)
>         err(1, "Cannot create file pftmp/foo.txt.");
> ```

Please remove the trailing dots (`.`) from `err()` messages -- `err()` function automatically adds the necessary chars.
___
*[LibOS/shim/test/regression/pf_rename.c, line 18 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3qQYz5gNhjNNytQbf:-Mj3qQYz5gNhjNNytQbg:b-fyjs7f) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L18)):*
> ```C
>         err(1, "Cannot create file pftmp/foo.txt.");
> 
>     int num_bytes = write(fd, input_text, strlen(input_text) + 1);
> ```

Please wrap the `write()` call in a while loop (to continue on partial writes). See example here: https://github.com/gramineproject/graphene/pull/2646/files#diff-7dcf98136ae60e9b2d915bf6e8548e9a3251dea52bbe975633cbc0ebbbeb2945
___
*[LibOS/shim/test/regression/pf_rename.c, line 24 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3qZoDD8lJfinMZUsf:-Mj3qZoDD8lJfinMZUsg:bopfwh9) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L24)):*
> ```C
>     }
> 
>     close(fd);
> ```

Please add error checking for `close()` functions too.
___
*[LibOS/shim/test/regression/pf_rename.c, line 28 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3r2mm9vWCb5BoZDZJ:-Mj3r2mm9vWCb5BoZDZK:bni24b3) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L28)):*
> ```C
>     int ret = rename("pftmp/foo.txt", "pftmp/bar.txt");
>     if (ret < 0) {
>         fd = open("pftmp/foo.txt", O_RDONLY);
> ```

To be honest, I don't see a point in this logic of "rename failed, let's verify that the original file is correct". You will never hit this logic in this test, because you don't explicitly make `rename()` above to fail (like, why would it fail?). So all this code is dead code. Let's just remove it (or you figure out a way to make `rename()` above fail).
___
*[LibOS/shim/test/regression/pf_rename.c, line 30 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3qsIy7IojoDZMUBho:-Mj3qsIy7IojoDZMUBhp:b-8g7ode) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L30)):*
> ```C
>         fd = open("pftmp/foo.txt", O_RDONLY);
>         if (fd < 0)
>             err(1, "Rename failed, Original file corrupted & unusable.");
> ```

`Original` -> `original`, here and below
___
*[LibOS/shim/test/regression/pf_rename.c, line 32 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3qocwAPVk-6p14zn_:-Mj3qocwAPVk-6p14zna:b-rtb9sp) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L32)):*
> ```C
>             err(1, "Rename failed, Original file corrupted & unusable.");
> 
>         num_bytes = read(fd, buf, sizeof(buf));
> ```

Please wrap the `read()` call in a while loop (to continue on partial reads). See example here: https://github.com/gramineproject/graphene/pull/2646/files#diff-7dcf98136ae60e9b2d915bf6e8548e9a3251dea52bbe975633cbc0ebbbeb2945
___
*[LibOS/shim/test/regression/test_libos.py, line 840 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3rgZp7yJwgvzz-6W7:-Mj3rgZp7yJwgvzz-6W8:bmxglif) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/test_libos.py#L840)):*
> ```Python
>                 os.remove("pftmp/bar.txt")
>         else:
>             os.mkdir("pftmp")
> ```

You can simplify it like this (didn't actually run, may contain small bugs):
```
for path in ['pftmp/foo.txt', 'pftmp/bar.txt']:
    if os.path.exists(path):
        os.remove(path)
os.makedirs('pftmp/', exist_ok=True)
```
___
*[Pal/src/host/Linux-SGX/db_files.c, line 128 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3x3D22xsV7EJuVp13:-Mj3x3D22xsV7EJuVp14:b-i2w8hs) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L128)):*
> ```C
>          * the metadata in the file, so open with RDWR mode with necessary share permissions.
>          * For normal files, reset this option.
>          */
> ```

The comment seems stale.

`If it is a protected file and` -- you can just remove this part. It is obvious here that we work on the PF.

`For normal files, reset this option.` -- stale comment, we don't do anything in this code snippet.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 797 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3xL7o10rcfcZWhxY6:-Mj3xL7o10rcfcZWhxY7:b-cx11ft) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L797)):*
> ```C
> }
> ```

Why this newline? Remove it.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 811 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yZunAYA0NTCNe7KW:-Mj3yZunAYA0NTCNe7KX:b-xim26t) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L811)):*
> ```C
>     /* TODO: Handle the case of renaming a file that has a file handle already open, so that the
>      * file operations work on both the handles properly
>      */
> ```

What is the scenario for this TODO, and why it doesn't work currently? I don't remember this...
___
*[Pal/src/host/Linux-SGX/db_files.c, line 850 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3xqey7hgpbxg2-SG8:-Mj3xqey7hgpbxg2-SG9:b-6lk23j) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L850)):*
> ```C
>     /* If the rename ocall fails, restore the metadata back to current file path, so that the
>      * file does not become unusable.
>      */
> ```

This comment is not needed because you already have another comment about it below. Please just remove this comment.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 854 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yA929fs1l3uZF0kP:-Mj3yA929fs1l3uZF0kQ:b3f66u1) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L854)):*
> ```C
>         free(tmp);
>         /* restore the original file name in pf metadata */
>         pf_status_t pf_ret =  pf_rename(pf->context, handle->file.realpath);
> ```

Remove double whitespace.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 856 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yEk9-t6IVuzbTgyD:-Mj3yEk9-t6IVuzbTgyE:bpjl6f) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L856)):*
> ```C
>         pf_status_t pf_ret =  pf_rename(pf->context, handle->file.realpath);
>         if (PF_FAILURE(pf_ret)) {
>             log_warning("rename failed: %s, the file might be unusable.", pf_strerror(pf_ret));
> ```

Please start with capital letter (`Rename...`) and remove the final dot -- just to be uniform with the other log message.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 857 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yP7P0Hf9StJTbQnX:-Mj3yP7P0Hf9StJTbQnY:b-pdu2r0) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L857)):*
> ```C
>         if (PF_FAILURE(pf_ret)) {
>             log_warning("rename failed: %s, the file might be unusable.", pf_strerror(pf_ret));
>             return -PAL_ERROR_DENIED;
> ```

Please remove this return. We will already return with the error code in the next line, so no need for this one.
___
*[Pal/src/host/Linux-SGX/protected-files/protected_files.c, line 376 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3w0-IBpQNbo_6Apax:-Mj3w0-IBpQNbo_6Apay:b-6bxizv) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/protected-files/protected_files.c#L376)):*
> ```C
> 
> static bool ipf_rename_file(pf_context_t* pf, const char* new_path) {
>     if (!new_path) {
> ```

Actually, can we move this check to `pf_rename()`? Sounds more logical to check it there.
___
*[Pal/src/host/Linux-SGX/protected-files/protected_files.c, line 1347 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3wQE-2SaIxlwFjFAW:-Mj3wQE-2SaIxlwFjFAX:b-pgmyh8) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/protected-files/protected_files.c#L1347)):*
> ```C
> 
>     if (new_path && strlen(new_path) > PATH_MAX_SIZE - 1) {
>         pf->last_error = PF_STATUS_PATH_TOO_LONG;
> ```

You don't need to set it here. Just remove this line. `pf->last_error` is a way to propagate the error to the upper levels of the PF implementation, but here you are already at the upmost level.


<!-- Sent from Reviewable.io -->

svenkata9 committed:
[Pal/Linux-SGX] Fix renaming issue with protected files

With certain workloads involving Federated Learning, the frameworks
create protected files and rename them as part of the flow. This does
not work currently with protected files because every protected file
maintains the path name in its metadata (stored when the file was first
created) and which is checked during open to make sure the incoming path
is the same as what is already stored in the metadata for this file.

A normal rename call does not need the file to be open in RW mode, but
since the metadata needs to be updated with the new path, we pass a
special PAL_OPTION_RENAME flag to DkStreamOpen during rename and open
the protected file in RW mode with required permissions, update the
metadata with the new path and flush it. Then the flow goes through the
one for normal file renaming. If the ocall to rename fails, we restore
the metadata back to what it was earlier so the file remains usable.

Signed-off-by: Sankaranarayanan Venkatasubramanian <sankaranarayanan.venkatasubramanian@intel.com>

svenkata9 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/gramineproject/graphene/2681)* status: 6 of 11 files reviewed, 22 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv and @svenkata9)

___
*[-- commits, line 5 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3pH2W93B098FdFtlV):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`renames` -> `rename`
</blockquote></details>

Done.
___
*[-- commits, line 15 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3p7yY-rO20zDfIwaQ):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

This doesn't seem to wrap on 72-char limit (the default limit for git messages). Could you re-wrap lines to 72 chars?
</blockquote></details>

Done.
___
*[LibOS/shim/test/regression/.gitignore, line 77 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3pX96A2ETRkiVe4r_:-Mj8ms9oAFcmJJ0AVgSy:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/.gitignore#L77)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Please move `/pftmp` (and also `/tmp`) in a separate section in this file (i.e., separates from other files by empty lines). These two entries are directories for temporary files, not executables, so they don't belong in this list logically.
</blockquote></details>

Done.
___
*[LibOS/shim/test/regression/pf_rename.c, line 14 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3pvgJB7TcKA0U1bRX:-Mj8mx06-TrYHWv7XMvc:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L14)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Can you use macros for these two paths? Like this:
```
#define PF_FOO_PATH "pftmp/foo.txt"
#define PF_BAR_PATH "pftmp/bar.txt"
```
</blockquote></details>

Done.
___
*[LibOS/shim/test/regression/pf_rename.c, line 16 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3q9MG5FpakVxcfCeH:-Mj8my-66p6hpxxBskbO:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L16)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Please remove the trailing dots (`.`) from `err()` messages -- `err()` function automatically adds the necessary chars.
</blockquote></details>

Done.
___
*[LibOS/shim/test/regression/pf_rename.c, line 18 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3qQYz5gNhjNNytQbf:-Mj8mzML5eNMNm-cT8DS:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L18)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Please wrap the `write()` call in a while loop (to continue on partial writes). See example here: https://github.com/gramineproject/graphene/pull/2646/files#diff-7dcf98136ae60e9b2d915bf6e8548e9a3251dea52bbe975633cbc0ebbbeb2945
</blockquote></details>

Done.
___
*[LibOS/shim/test/regression/pf_rename.c, line 24 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3qZoDD8lJfinMZUsf:-Mj8n07zCYdSPFKSGWpg:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L24)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Please add error checking for `close()` functions too.
</blockquote></details>

Done.
___
*[LibOS/shim/test/regression/pf_rename.c, line 28 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3r2mm9vWCb5BoZDZJ:-Mj8n1Dq6ctiNIX-Cfcg:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L28)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

To be honest, I don't see a point in this logic of "rename failed, let's verify that the original file is correct". You will never hit this logic in this test, because you don't explicitly make `rename()` above to fail (like, why would it fail?). So all this code is dead code. Let's just remove it (or you figure out a way to make `rename()` above fail).
</blockquote></details>

Done.
___
*[LibOS/shim/test/regression/pf_rename.c, line 30 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3qsIy7IojoDZMUBho:-Mj8n1rx2jISe8RrByxy:blrygtm) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L30)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`Original` -> `original`, here and below
</blockquote></details>

Removed the code. So, this is NA
___
*[LibOS/shim/test/regression/pf_rename.c, line 32 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3qocwAPVk-6p14zn_:-Mj8n5Ae8oKouEBfprA6:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/pf_rename.c#L32)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Please wrap the `read()` call in a while loop (to continue on partial reads). See example here: https://github.com/gramineproject/graphene/pull/2646/files#diff-7dcf98136ae60e9b2d915bf6e8548e9a3251dea52bbe975633cbc0ebbbeb2945
</blockquote></details>

Done.
___
*[LibOS/shim/test/regression/test_libos.py, line 840 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3rgZp7yJwgvzz-6W7:-Mj8nG_K7qh5mq0Vc3AQ:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/LibOS/shim/test/regression/test_libos.py#L840)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

You can simplify it like this (didn't actually run, may contain small bugs):
```
for path in ['pftmp/foo.txt', 'pftmp/bar.txt']:
    if os.path.exists(path):
        os.remove(path)
os.makedirs('pftmp/', exist_ok=True)
```
</blockquote></details>

Done.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 128 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3x3D22xsV7EJuVp13:-Mj8nIW7BrDTjU6cZ2I-:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L128)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

The comment seems stale.

`If it is a protected file and` -- you can just remove this part. It is obvious here that we work on the PF.

`For normal files, reset this option.` -- stale comment, we don't do anything in this code snippet.
</blockquote></details>

Done.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 797 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3xL7o10rcfcZWhxY6:-Mj8nNDV5tmj285TQsn8:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L797)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Why this newline? Remove it.
</blockquote></details>

Done.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 811 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yZunAYA0NTCNe7KW:-Mj8UG9LCXJIXOAfzZew:bshhr8h) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L811)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

What is the scenario for this TODO, and why it doesn't work currently? I don't remember this...
</blockquote></details>

https://reviewable.io/reviews/gramineproject/graphene/2521#-Mi6_Ynm5qI5oJuEvpt8
___
*[Pal/src/host/Linux-SGX/db_files.c, line 850 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3xqey7hgpbxg2-SG8:-Mj8nTJN8f2na31F-WcN:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L850)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

This comment is not needed because you already have another comment about it below. Please just remove this comment.
</blockquote></details>

Done.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 856 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yEk9-t6IVuzbTgyD:-Mj8nUBD8YUbLiU24zMa:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L856)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Please start with capital letter (`Rename...`) and remove the final dot -- just to be uniform with the other log message.
</blockquote></details>

Done.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 857 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yP7P0Hf9StJTbQnX:-Mj8nVCq7TfFbX8Mz3Re:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L857)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Please remove this return. We will already return with the error code in the next line, so no need for this one.
</blockquote></details>

Done.
___
*[Pal/src/host/Linux-SGX/protected-files/protected_files.c, line 376 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3w0-IBpQNbo_6Apax:-Mj8nXBC-BjDpz-K3CWZ:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/protected-files/protected_files.c#L376)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Actually, can we move this check to `pf_rename()`? Sounds more logical to check it there.
</blockquote></details>

Done.
___
*[Pal/src/host/Linux-SGX/protected-files/protected_files.c, line 1347 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3wQE-2SaIxlwFjFAW:-Mj8nabM91Qj2K75cY0K:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/protected-files/protected_files.c#L1347)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

You don't need to set it here. Just remove this line. `pf->last_error` is a way to propagate the error to the upper levels of the PF implementation, but here you are already at the upmost level.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

dimakuv reviewed (changes requested):
> Reviewed 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/graphene/2681)* status: all files reviewed, 11 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @svenkata9)

___
*[-- commits, line 5 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3pH2W93B098FdFtlV):*
<details><summary><i>Previously, svenkata9 (Sankaranarayanan Venkatasubramanian) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

That's not what I meant. I was expecting you to `git commit --amend` your first commit, instead you have two commits now: the first one with the old text, and the second one (which should have been a fixup commit) with the new text.

Let's keep it as-is, I'll fix it during final rebase. But for the future, changing commit messages is only meamingful using the `git rebase` or `git commit --amend` flows.
___
*[LibOS/shim/test/regression/pf_rename.c, line 23 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjEZg2v9XxHAzo9JwGT:-MjEZg2v9XxHAzo9JwGU:bhmjw3p) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L23)):*
> ```C
>         err(1, "File creation failed");
> 
>     int size = strlen(input_text) + 1;
> ```

`int` -> `size_t`
___
*[LibOS/shim/test/regression/pf_rename.c, line 57 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_-eNBC_zJfajXRLB:-MjE_-eNBC_zJfajXRLC:bx3gbmw) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L57)):*
> ```C
>         if (n == 0) {
>             if (size > 0) {
>                 warnx("read less bytes than expected");
> ```

`read` -> `Read`
___
*[LibOS/shim/test/regression/pf_rename.c, line 65 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_9kj3roQq78jWsKB:-MjE_9kj3roQq78jWsKC:bmb3weo) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L65)):*
> ```C
>     } while (pos < BUF_SIZE);
> 
>     close(fd);
> ```

You forgot `ret = close(fd)`
___
*[LibOS/shim/test/regression/pf_rename.c, line 76 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_K94-lLa13y6SWPd:-MjE_K94-lLa13y6SWPe:b-ybe39g) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L76)):*
> ```C
> 
>     printf("TEST OK\n");
> ```

Remove this empty line.
___
*[LibOS/shim/test/regression/pf_rename.c, line 79 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_IidD5xJtAvEoPuK:-MjE_IidD5xJtAvEoPuL:b-ybe39g) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L79)):*
> ```C
>     return 0;
> }
> ```

Remove this empty line.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 811 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yZunAYA0NTCNe7KW:-MjE_o6f86_whH5kD4Fa:b-4ekr2m) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L811)):*
<details><summary><i>Previously, svenkata9 (Sankaranarayanan Venkatasubramanian) wrote…</i></summary><blockquote>

https://reviewable.io/reviews/gramineproject/graphene/2521#-Mi6_Ynm5qI5oJuEvpt8
</blockquote></details>

OK. Thanks.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 856 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yEk9-t6IVuzbTgyD:-MjE_v4G22mpb0BmeruU:b6g6vu6) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L856)):*
<details><summary><i>Previously, svenkata9 (Sankaranarayanan Venkatasubramanian) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Forgot to remove the final dot :)
___
*[Pal/src/host/Linux-SGX/db_files.c, line 127 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_S3qBhBqSpiXXY0_:-MjE_S3rEvd4ujdcOQ3U:bsotk7) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/Pal/src/host/Linux-SGX/db_files.c#L127)):*
> ```C
>         /* The file is being opened for renaming. We will need to update the metadata in the file,
>          * so open with RDWR mode with necessary share permissions.
>          */
> ```

Please move this `*/` to the end of the previous line (personal preference, but we do it almost everywhere in our codebase)
___
*[Pal/src/host/Linux-SGX/db_files.c, line 809 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_ptU5TKW1bEGs9wK:-MjE_ptU5TKW1bEGs9wL:bsotk7) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/Pal/src/host/Linux-SGX/db_files.c#L809)):*
> ```C
>     /* TODO: Handle the case of renaming a file that has a file handle already open, so that the
>      * file operations work on both the handles properly
>      */
> ```

Please move this `*/` to the end of the previous line (personal preference, but we do it almost everywhere in our codebase)


<!-- Sent from Reviewable.io -->

pwmarcz reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/gramineproject/graphene/2681)* status: all files reviewed, 12 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv and @svenkata9)

*[a discussion](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yhSE7OocZItNkL4P:-Mj9-ktS1ThYrNdIi0So:b-7tmu2c) (no related file):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

@pwmarcz will need to review this. Especially the part with the new `rename` test -- Pawel also added a similar test in his PR, so I think the two tests can be merged into one.
</blockquote></details>

I hoped I would be able to point to the new repo, but we have some technical issues with Reviewable and could not merge the new branch yet...

Anyway, the PR (https://github.com/gramineproject/gramine/pull/5) already contains a test suite for rename; it's skipped for protected files but you can easily un-skip it if you want to check your code.

---
___
*[LibOS/shim/test/regression/pf_rename.c, line 1 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjFObr45b4OqvddMCxl:-MjFObr45b4OqvddMCxm:b1y1co0) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L1)):*
> ```C
> /* Protected file renaming. Renaming a file without closing its handle. */
> ```

This comment says that the test is "Renaming a file without closing its handle.", but the test is actually closing the handle and then renaming the file. Why? I think _renaming a file without closing its handle_ is actually the right thing to test here.

Anyway, as mentioned elsewhere, you can try using the test from https://github.com/gramineproject/gramine/pull/5 (hopefully soon to be merged). The `chroot/fs.c` file has been rewritten, but since this PR changes only one line in it, the change should not be hard to adapt.


<!-- Sent from Reviewable.io -->

svenkata9 committed:
fixup! [Pal/Linux-SGX] Fix renaming issue with protected files

svenkata9 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/gramineproject/graphene/2681)* status: 9 of 11 files reviewed, 12 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ), "fixup! " found in commit messages' one-liners (waiting on @dimakuv, @pwmarcz, and @svenkata9)

*[a discussion](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yhSE7OocZItNkL4P:-MjSmEjwDNrWI8WIIipd:b-xhlcka) (no related file):*
<details><summary><i>Previously, pwmarcz (Paweł Marczewski) wrote…</i></summary><blockquote>

I hoped I would be able to point to the new repo, but we have some technical issues with Reviewable and could not merge the new branch yet...

Anyway, the PR (https://github.com/gramineproject/gramine/pull/5) already contains a test suite for rename; it's skipped for protected files but you can easily un-skip it if you want to check your code.
</blockquote></details>

Thanks. I can work on this after this PR gets merged

---
___
*[-- commits, line 5 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3pH2W93B098FdFtlV):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

That's not what I meant. I was expecting you to `git commit --amend` your first commit, instead you have two commits now: the first one with the old text, and the second one (which should have been a fixup commit) with the new text.

Let's keep it as-is, I'll fix it during final rebase. But for the future, changing commit messages is only meamingful using the `git rebase` or `git commit --amend` flows.
</blockquote></details>

I see. I misunderstood the sequence.
___
*[LibOS/shim/test/regression/pf_rename.c, line 1 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjFObr45b4OqvddMCxl:-MjSmONS7V3RmxOgCetM:b-svnu5y) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L1)):*
<details><summary><i>Previously, pwmarcz (Paweł Marczewski) wrote…</i></summary><blockquote>

This comment says that the test is "Renaming a file without closing its handle.", but the test is actually closing the handle and then renaming the file. Why? I think _renaming a file without closing its handle_ is actually the right thing to test here.

Anyway, as mentioned elsewhere, you can try using the test from https://github.com/gramineproject/gramine/pull/5 (hopefully soon to be merged). The `chroot/fs.c` file has been rewritten, but since this PR changes only one line in it, the change should not be hard to adapt.

</blockquote></details>

Thanks. I will update the code once your PR is merged. For now, I will keep it like this.

Like I indicated, renaming a protected file without closing the handle needs to be worked out in a separate PR. There are few problems that need to be ironed out. (1) the g_protected_files uses a single key UT, can can return a wrong output when find_protected_file(handle) is used. Further, if the file is opened in write mode, it will fail with the latest code because we allow only one write fd, and rename opens it in RDWR mode.

For the current issue, let us keep it like this to unblock the customer who is waiting for this change.
___
*[LibOS/shim/test/regression/pf_rename.c, line 23 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjEZg2v9XxHAzo9JwGT:-MjSn7fh3BTLeUiThXcz:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L23)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`int` -> `size_t`
</blockquote></details>

Done.
___
*[LibOS/shim/test/regression/pf_rename.c, line 57 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_-eNBC_zJfajXRLB:-MjSn8gm6wqv5-hJLTqH:b-smqjzj) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L57)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

`read` -> `Read`
</blockquote></details>

Done. But, I just reused from the reference you pointed out. It is all using lower case error message starts. https://github.com/gramineproject/graphene/pull/2646/files#
___
*[LibOS/shim/test/regression/pf_rename.c, line 65 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_9kj3roQq78jWsKB:-MjSnmp6CdqSXI_2gL0j:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L65)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

You forgot `ret = close(fd)`
</blockquote></details>

Done.
___
*[LibOS/shim/test/regression/pf_rename.c, line 76 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_K94-lLa13y6SWPd:-MjSnnhqBMu_vIFB1VY6:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L76)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Remove this empty line.
</blockquote></details>

Done.
___
*[LibOS/shim/test/regression/pf_rename.c, line 79 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_IidD5xJtAvEoPuK:-MjSno401FZdFY2K_c2d:bzd2x1b) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L79)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Remove this empty line.
</blockquote></details>

Done, but I follow leaving a new line at the end (I guess C11 still has that mention, though the compilers work even when we don't leave a new line at the end).
___
*[Pal/src/host/Linux-SGX/db_files.c, line 856 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yEk9-t6IVuzbTgyD:-MjSo8fX2_IlMMtzB8dg:b-896fix) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L856)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Forgot to remove the final dot :)
</blockquote></details>

Done.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 127 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_S3qBhBqSpiXXY0_:-MjSo-mZDjNdpXz4pOY0:bgul9sk) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/Pal/src/host/Linux-SGX/db_files.c#L127)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Please move this `*/` to the end of the previous line (personal preference, but we do it almost everywhere in our codebase)
</blockquote></details>

Done. But, appearance-wise, having it in the next line looks good. Anyway, I'll follow what the majority follow.
___
*[Pal/src/host/Linux-SGX/db_files.c, line 809 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_ptU5TKW1bEGs9wK:-MjSo7sq806ubVPf0nRN:bgul9sk) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/Pal/src/host/Linux-SGX/db_files.c#L809)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Please move this `*/` to the end of the previous line (personal preference, but we do it almost everywhere in our codebase)
</blockquote></details>

Done. But, appearance-wise, having it in the next line looks good. Anyway, I'll follow what the majority follow.


<!-- Sent from Reviewable.io -->

svenkata9 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/gramineproject/graphene/2681)* status: 9 of 11 files reviewed, 12 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ), "fixup! " found in commit messages' one-liners (waiting on @dimakuv, @duanbing, and @pwmarcz)

___
*[Pal/src/host/Linux-SGX/db_files.c, line 860 at r1](https://reviewable.io/reviews/gramineproject/graphene/2681#-MizImmysnxx-MvrsSf7-r1-860:-MjSoJv77jzktrDYQ1-9:b-zcjvte) ([raw file](https://github.com/gramineproject/graphene/blob/9c71eeb5f4adfaecc73e50469c3e0cbf5c6f5570/Pal/src/host/Linux-SGX/db_files.c#L860)):*
<details><summary><i>Previously, duanbing (Bing) wrote…</i></summary><blockquote>

Got it. Thanks.
</blockquote></details>

Thanks


<!-- Sent from Reviewable.io -->

dimakuv reviewed (changes requested):
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/graphene/2681)* status: all files reviewed, 4 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (2 more required, approved so far: ), "fixup! " found in commit messages' one-liners (waiting on @dimakuv, @duanbing, @pwmarcz, and @svenkata9)

*[a discussion](https://reviewable.io/reviews/gramineproject/graphene/2681#-Mj3yhSE7OocZItNkL4P:-MjT0qKa6DT-SD3X-ela:bs3knt9) (no related file):*
<details><summary><i>Previously, svenkata9 (Sankaranarayanan Venkatasubramanian) wrote…</i></summary><blockquote>

Thanks. I can work on this after this PR gets merged
</blockquote></details>

https://github.com/gramineproject/gramine/pull/5 -- the PR was merged.

I suggest to now rebase your PR locally, check that the tests are working for your cases, and if everything works as expected, push your rebase to GitHub so everyone can see it. Maybe it will even make more sense to recreate the PR in the new repository now.

---
___
*[LibOS/shim/test/regression/pf_rename.c, line 57 at r2](https://reviewable.io/reviews/gramineproject/graphene/2681#-MjE_-eNBC_zJfajXRLB:-MjT1bsV32G4B2ZK3iZA:b-pt2p37) ([raw file](https://github.com/gramineproject/graphene/blob/2a00c5b697be60f6bf6f0171829e750ebe172b58/LibOS/shim/test/regression/pf_rename.c#L57)):*
<details><summary><i>Previously, svenkata9 (Sankaranarayanan Venkatasubramanian) wrote…</i></summary><blockquote>

Done. But, I just reused from the reference you pointed out. It is all using lower case error message starts. https://github.com/gramineproject/graphene/pull/2646/files#
</blockquote></details>

OK. I understand your point, but generally we don't have a strong preference on lower-case vs upper-case. It's just a matter of uniformity with the surrounding code. You have upper-case messages in all other places in this file, so we should be uniform and also have upper-case here.


<!-- Sent from Reviewable.io -->

mkow commented:
This was superseded by https://github.com/gramineproject/gramine/pull/31, closing.
